amazon-kinesis-client-multilang/src/test/java/software/amazon/kinesis/multilang/MultiLangDaemonConfigTest.java

```diff
@@ -48,39 +58,170 @@ public class MultiLangDaemonConfigTest {
     @Mock
     private KinesisClientLibConfigurator configurator;
 
-    @Before
-    public void setup() {
-        ConvertUtilsBean convertUtilsBean = new ConvertUtilsBean();
-        BeanUtilsBean utilsBean = new BeanUtilsBean(convertUtilsBean);
-        MultiLangDaemonConfiguration multiLangDaemonConfiguration = new MultiLangDaemonConfiguration(utilsBean,
-                convertUtilsBean);
-        multiLangDaemonConfiguration.setApplicationName("cool-app");
-        multiLangDaemonConfiguration.setStreamName("cool-stream");
-        multiLangDaemonConfiguration.setWorkerIdentifier("cool-worker");
+    public void setup(String streamName, String streamArn) {
+
+        String PROPERTIES = String.format("executableName = %s\n"
+                + "applicationName = %s\n"
+                + "AWSCredentialsProvider = DefaultAWSCredentialsProviderChain\n"
+                + "processingLanguage = malbolge\n"
+                + "regionName = %s\n",
+                TestExe,
+                TestApplicationName,
+                TestRegion);
+
+        if(streamName != null){
+            PROPERTIES += String.format("streamName = %s\n", streamName);
+        }
+        if(streamArn != null){
+            PROPERTIES += String.format("streamArn = %s\n", streamArn);
+        }
+        classLoader = Mockito.mock(ClassLoader.class);
+
+        Mockito.doReturn(new ByteArrayInputStream(PROPERTIES.getBytes())).when(classLoader)
+                .getResourceAsStream(FILENAME);
+
         when(credentialsProvider.resolveCredentials()).thenReturn(creds);
         when(creds.accessKeyId()).thenReturn("cool-user");
-        when(configurator.getConfiguration(any(Properties.class))).thenReturn(multiLangDaemonConfiguration);
+        configurator = new KinesisClientLibConfigurator();
+
     }
 
     @Test
-    public void constructorTest() throws IOException {
-        String PROPERTIES = "executableName = randomEXE \n" + "applicationName = testApp \n"
-                + "streamName = fakeStream \n" + "AWSCredentialsProvider = DefaultAWSCredentialsProviderChain\n"
-                + "processingLanguage = malbolge";
-        ClassLoader classLoader = Mockito.mock(ClassLoader.class);
+    public void testConstructorFailsBecauseStreamArnIsInvalid() throws IOException {
+        setup("", "this_is_not_a_valid_arn");
 
-        Mockito.doReturn(new ByteArrayInputStream(PROPERTIES.getBytes())).when(classLoader)
-                .getResourceAsStream(FILENAME);
+        assertThrows(Exception.class, () -> new MultiLangDaemonConfig(FILENAME, classLoader, configurator));
```

Comment:
Is it possible to assert a more specific exception other than `Exception.class`?

Reply:
Yes

Reply:
KISS and leverage the test framework:
```
@Test(expected = MyException.class)
public void test... throws IOException {
   setup(...);
   new MLDC(...);
}
```

This applies elsewhere, as well.

---
DRY: unless I overlooked something, every invocation of `setup(...)` is followed w/ an instantiation of MLDC. The instantiation of MLDC could be in `setup(...)`